aio/content/examples/dynamic-component-loader/src/app/ad.service.ts

```diff
@@ -1,30 +1,32 @@
 // #docregion
-import { Injectable } from '@angular/core';
+import { Injectable, Type } from '@angular/core';
 
-import { HeroJobAdComponent } from './hero-job-ad.component';
-import { HeroProfileComponent } from './hero-profile.component';
-import { AdItem } from './ad-item';
-
-@Injectable()
+@Injectable({ providedIn: 'root' })
 export class AdService {
   getAds() {
     return [
-      new AdItem(
-        HeroProfileComponent,
-        { name: 'Bombasto', bio: 'Brave as they come' }
-      ),
-      new AdItem(
-        HeroProfileComponent,
-        { name: 'Dr. IQ', bio: 'Smart as they come' }
-      ),
-      new AdItem(
-        HeroJobAdComponent,
-        { headline: 'Hiring for several positions', body: 'Submit your resume today!' }
-      ),
-      new AdItem(
-        HeroJobAdComponent,
-        { headline: 'Openings in all departments', body: 'Apply today' }
-      )
-    ];
+      {
+        component: import('./hero-profile.component').then(m => m.HeroProfileComponent),
```

Comment:
I think we should we wrap dynamic imports into a function, so that we don't invoke them right away:
```suggestion
        component: () => import('./hero-profile.component').then(m => m.HeroProfileComponent),
```

Reply:
I'm running into an issue with the function. 😕
It's called continusly, likely due to the implementation of the ngComponentOutlet inputs. Remember the function is called by the template.

Reply:
I think that calling the function multiple times is fine, since there should be no extra requests made (if network cache is not disabled), i.e. you'll just get a reference to a class by calling the function. Please let me know if you see a different behavior that we'd need to investigate.

Reply:
The whole app becomes unresponsive, there is a CD loop because of the Async pipe. (CD => function gets called => Async Pipe triggers CD)

Reply:
@JeanMeche thanks for the info and sorry for the delay in answering. I'd propose to avoid using dynamic imports for now, so that we can land the change (since it simplifies the guide a lot!) and get back to this question as a followup.

Reply:
Updated as requested ✅